packages/graphql/src/translate/create-auth-and-params.ts

```diff
@@ -102,13 +106,19 @@ function createAuthPredicate({
                     paramValue = dotProp.get({ value: context }, `value.${ctxPath}`) as string;
                 }
 
-                if (paramValue === undefined) {
+                if (paramValue === undefined && allowUnauthenticated !== true) {
                     throw new Neo4jGraphQLAuthenticationError("Unauthenticated");
                 }
 
-                const param = `${chainStr}_${key}`;
-                res.params[param] = paramValue;
-                res.strs.push(`EXISTS(${varName}.${key}) AND ${varName}.${key} = $${param}`);
+                if (paramValue === undefined) {
+                    res.strs.push("false");
+                } else if (paramValue === null) {
+                    res.strs.push(`${varName}.${key} IS null`);
```

Comment:
Real nitpicky, but could we capitalise `null` here?

Reply:
Sure, I have replaced that.
This specific condition is required as an issue appear if we perform a `null` comparison with `=`, the result would be `null` and not a boolean. This caused APOC.validate to raise an exception. https://community.neo4j.com/t/neo-clienterror-procedure-procedurecallfailed-with-apoc-do-when/14017/12